lib/array_equals.rb

```diff
@@ -2,4 +2,51 @@
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
   raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return true
+  elsif array1 == [] && array2 == []
+    return true
+  elsif array1 == nil || array2 == nil
```

Comment:
The statements on line 9 & 10 seem contradictory to the statement on line 5 & 6. Could you please look into it?